src/credentials/GroupDataProvider.h

```diff
@@ -44,111 +45,138 @@ class GroupDataProvider
     // `endpoint` value.
     struct GroupMapping
     {
+        static constexpr size_t kGroupNameMax = CHIP_CONFIG_MAX_GROUP_NAME_LENGTH;
+
         // The endpoint to which a GroupId is mapped.
-        EndpointId endpoint = 0;
+        EndpointId endpoint = kInvalidEndpointId;
         // The GroupId, which, when received in a message will map to the `endpoint`.
-        GroupId group = 0;
+        GroupId group = kInvalidEndpointId;
         // Group name
-        CharSpan name;
+        char name[kGroupNameMax + 1] = { 0 };
 
         GroupMapping() = default;
-        GroupMapping(EndpointId eid, GroupId gid) : endpoint(eid), group(gid) {}
-        GroupMapping(EndpointId eid, GroupId gid, const CharSpan & groupName) : endpoint(eid), group(gid), name(groupName) {}
-        GroupMapping(EndpointId eid, GroupId gid, const char * groupName) :
-            endpoint(eid), group(gid), name(CharSpan(groupName, strnlen(groupName, CHIP_CONFIG_MAX_GROUP_NAME_LENGTH)))
-        {}
+        GroupMapping(EndpointId eid, GroupId gid) : GroupMapping(eid, gid, nullptr) {}
+        GroupMapping(EndpointId eid, GroupId gid, const char * groupName) : endpoint(eid), group(gid)
+        {
+            if (nullptr == groupName)
+            {
+                name[0] = 0;
+            }
+            else
+            {
+                strncpy(name, groupName, kGroupNameMax);
```

Comment:
We expect a null-terminated string, so we should null-terminate every time. This applies to every strncpy.

Overall, perhaps it would be safer to have a single-byte size storage always kept, and just always encde CharSpans. Otherwise, everything you do `strn*`, you need to check whether the string had been null-terminated, and to add forceful nul-termination on loading from storage.

Reply:
I have explicitly set the null termination within the limits of the name field.